docs/advanced/api-v2/rest.md

```diff
@@ -0,0 +1,4019 @@
+---
+title: REST API V2
+---
+
+FORMAT: 1A
+HOST: https://api.counterparty.io
+
+# Counterparty Core API
+
+The Counterparty Core API is the recommended (and only supported) way to query the state of a Counterparty node. 
+
+API routes are divided into 11 groups:
+- [`/blocks`](#group-blocks)
+- [`/transactions`](#group-transactions)
+- [`/addresses`](#group-addresses)
+- [`/assets`](#group-assets)
+- [`/orders`](#group-orders)
+- [`/bets`](#group-bets)
+- [`/dispensers`](#group-dispensers)
+- [`/burns`](#group-burns)
+- [`/events`](#group-events)
+- [`/mempool`](#group-mempool)
+- [`/backend`](#group-backend)
+
+Notes:
+
+- When the server is not ready, that is to say when all the blocks are not yet parsed, all routes return a 503 error except `/` and those in the `/blocks`, `/transactions` and `/backend` groups which always return a result.
+
+- All API responses contain the following 3 headers:
+
+    * `X-COUNTERPARTY-HEIGHT` contains the last block parsed by Counterparty
+    * `X-BACKEND-HEIGHT` contains the last block known to Bitcoin Core
+    * `X-COUNTERPARTY-READY` contains true if `X-COUNTERPARTY-HEIGHT` >= `X-BACKEND-HEIGHT` - 1
```

Comment:
What happens if bitcoind is still catching up?

Reply:
ach sorry I don't follow. can you say that again?

Reply:
No problem what I said is wrong anyway ;-) for now as long as Addrindexrs is catching up the API server is not starting..

Reply:
Ahh I think we want it to run either way, no? It'd be nice if it gave a helpful error while bitcoind is stupidly completely unavailable.

Reply:
yeah I agree.. but it's not trivial.. the problem comes from Addrindexrs which does not start the RPC server until it has caught up.. this means that we cannot I don't even have the block count.. and also the connection step to Addrindexrs is blocking.. I will try to find a clean solution..

Reply:
ok done here https://github.com/CounterpartyXCP/counterparty-core/pull/1729/commits/9d61360bad2af407fa9e504948bef9c18a2f31bc
Now the API v2 starts before connecting to addrindexrs.. and as long it's not connected it returns a 503 error with a "Backend still not ready" message.